src/console_cmd.c

```diff
@@ -276,78 +416,1301 @@ TbBool cmd_exec(PlayerNumber plyr_idx, char *msg)
+            command_add_value(Cmd_TRAP_AVAILABLE, plyr_idx, id, 1, num);
             update_trap_tab_to_config();
             message_add(plyr_idx, "done!");
             return true;
-        } else if (strcmp(parstr, "give.door") == 0)
+        } 
+        else if (strcasecmp(parstr, "door") == 0)
+        {
+            if (pr2str != NULL)
+            {
+                message_add_fmt(plyr_idx, "%s", door_code_name(atoi(pr2str)));
+                return true;                
+            }
+        }
+        else if ( (strcasecmp(parstr, "give.door") == 0) || (strcasecmp(parstr, "door.give") == 0) )
         {
-            int id = atoi(pr2str);
+            long id = get_rid(door_desc, pr2str);
+            if (id <= 0)
+            {
+                if (strcasecmp(pr2str, "wooden") == 0)
+                {
+                    id = 1;
+                }
+                else if (strcasecmp(pr2str, "iron") == 0)
+                {
+                    id = 3;
+                }
+                else if (strcasecmp(pr2str, "magical") == 0)
+                {
+                    id = 4;
+                }
+                else
+                {
+                    id = atoi(pr2str);
+                }
+            }
             if (id <= 0 || id > trapdoor_conf.door_types_count)
                 return false;
-            script_process_value(Cmd_DOOR_AVAILABLE, plyr_idx, id, 1, 1);
+            unsigned char num = (pr3str != NULL) ? atoi(pr3str) : 1;
+            script_process_value(Cmd_DOOR_AVAILABLE, plyr_idx, id, 1, num);
             update_trap_tab_to_config();
             message_add(plyr_idx, "done!");
             return true;
         }
-    }
-    return false;
+        else if (strcasecmp(parstr, "room.available") == 0)
+        {
+            long roomid = get_rid(room_desc, pr2str);
+            if (roomid <= 0)
+            {
+                if (strcasecmp(pr2str, "Hatchery" ) == 0)
+                {
+                    roomid = RoK_GARDEN;
+                }
+                else if ( (strcasecmp(pr2str, "Guard" ) == 0) || (strcasecmp(pr2str, "GuardPost" ) == 0) )
+                {
+                    roomid = RoK_GUARDPOST;
+                }
+                else
+                {
+                    roomid = atoi(pr2str);
+                }
+            }
+            unsigned char available = (pr3str == NULL) ? 1 : atoi(pr3str);
+            PlayerNumber id = get_player_number_for_command(pr4str);
+            script_process_value(Cmd_ROOM_AVAILABLE, id, roomid, (TbBool)available, (TbBool)available);
+            update_room_tab_to_config();
+            return true;
+        }
+        else if ( (strcasecmp(parstr, "power.give") == 0) || (strcasecmp(parstr, "spell.give") == 0) )
+        {
+            long power = get_rid(power_desc, pr2str);
+            if (power <= 0)
+            {
+                if ( (strcasecmp(pr2str, "Imp" ) == 0) || (strcasecmp(pr2str, "CresteImp" ) == 0) )
```

Comment:
I think this should be 'create imp'

Reply:
The game treats separate words as separate parameters.

Reply:
you have 'creste' instead of 'create'